x-pack/plugins/security/public/management/roles/roles_grid/roles_grid_page.test.tsx

```diff
@@ -45,7 +45,9 @@ describe('<RolesGridPage />', () => {
   let history: ScopedHistory;
 
   beforeEach(() => {
-    history = (scopedHistoryMock.create() as unknown) as ScopedHistory;
+    history = (scopedHistoryMock.create({
+      createHref: jest.fn((location) => location.pathname!),
+    }) as unknown) as ScopedHistory;
```

Comment:
We don't usually add API to override core mock properties, as the same goal can be achieved by using the existing mocked methods.

```ts
const history: ScopedHistoryMock = scopedHistoryMock.create()
history.createHref.mockImplementation((location) => location.pathname!));
```

However I see that our `ScopedHistoryMock` is not compatible with `ScopedHistory`, which complicates the process a little, see:

```ts
const h: ScopedHistory =  scopedHistoryMock.create()

Type 'Mocked<Pick<ScopedHistory<unknown>, "replace" | "length" | "push" | "block" | "location" | "action" | "goBack" | "go" | "goForward" | "listen" | "createHref" | "createSubHistory">>' is missing the following properties from type 'ScopedHistory<unknown>': isActive, listeners, locationKeys, currentLocationKeyIndex, and 7 more.
```

@joshdover wdyt? Is this change fine with you, or should we 'fix' the mock instead? I guess that would force us to create a `IScopedHistory`, but OTOH the mock cannot currently be properly used for testing (appart from using it for `jest.mock('src/core/public')`)

Reply:
I agree we should probably fix the mock and/or create an interface that doesn't require this private properties. Can be done in a follow up PR though if @thomheymann wants to keep this PR focused or have someone Platform do it.

Reply:
Ok. I'll create an issue for the mock fix, and revert this change then.

Reply:
created https://github.com/elastic/kibana/issues/71166